core/src/mindustry/world/blocks/production/GenericCrafter.java

```diff
@@ -154,6 +175,24 @@ public void drawLight(){
             drawer.drawLight(this);
         }
 
+        @Override
+        public void drawSelect(){
+            super.drawSelect();
+            if(outputLiquids != null){
```

Comment:
This should be refactored into a separate block function that takes a raw x and y, because you're duplicating a lot of code here.